Emby.Server.Implementations/EntryPoints/LibraryChangedNotifier.cs

```
@@ -118,35 +118,34 @@ private void OnProviderRefreshProgress(object? sender, GenericEventArgs<Tuple<Ba
 
-        var dict = new Dictionary<string, string>();
-        dict["ItemId"] = item.Id.ToString("N", CultureInfo.InvariantCulture);
-        dict["Progress"] = progress.ToString(CultureInfo.InvariantCulture);
-
-        try
-        {
-            _sessionManager.SendMessageToAdminSessions(SessionMessageType.RefreshProgress, dict, CancellationToken.None);
-        }
-        catch
-        {
-        }
-
-        var collectionFolders = _libraryManager.GetCollectionFolders(item);
-
-        foreach (var collectionFolder in collectionFolders)
-        {
-            var collectionFolderDict = new Dictionary<string, string>
+        // Code can be simplified if calling item.GetRefreshProgress() returns the same progress value as the argument provided
+        // Push item & collectionFolders into a List<BaseItem>, iterate over them and create Dictionary when calling SendMessageToAdminSessions
+        List<Dictionary<string, string>> progressMessages = [
+            new Dictionary<string, string>
+            {
+                ["ItemId"] = item.Id.ToString("N", CultureInfo.InvariantCulture),
+                ["Progress"] = progress.ToString(CultureInfo.InvariantCulture)
+            },
+            .. _libraryManager.GetCollectionFolders(item).Select(collectionFolder => new Dictionary<string, string>()
             {
                 ["ItemId"] = collectionFolder.Id.ToString("N", CultureInfo.InvariantCulture),
                 ["Progress"] = (collectionFolder.GetRefreshProgress() ?? 0).ToString(CultureInfo.InvariantCulture)
-            };
+            }),
+        ];
 
-            try
-            {
-                _sessionManager.SendMessageToAdminSessions(SessionMessageType.RefreshProgress, collectionFolderDict, CancellationToken.None);
-            }
-            catch
+        foreach (var progressMessage in progressMessages)
+        {
+            // Explicit Fire & Forget
+            _ = Task.Run(async () =>
```

Never run a task like this as it will hide all exceptions and displaces the callstack making any errors untraceable

---

I was hesitant of making this change but the original caller is synchronously;
" An exception thrown in an unawaited Task won't be caught by the caller. "

That's why I wrapped it in the Task.Run, if in the future, there is a need to add logging to this call and to avoid having the exception being thrown at some random thread.
https://dev.to/souvikk27/fire-and-forget-strategy-for-non-blocking-long-running-tasks-5b1o

Some additional googling gave me:
[HandleSafeFireAndForget](https://github.com/TheCodeTraveler/AsyncAwaitBestPractices/blob/main/src/AsyncAwaitBestPractices/SafeFireAndForgetExtensions.shared.cs)

which is another possibility of wrapping it in an async void method.

---

Exactly, there is no await in the calling code.
Following the last part of the [docs](https://learn.microsoft.com/en-us/dotnet/standard/parallel-programming/exception-handling-task-parallel-library#observing-exceptions-by-using-the-taskexception-property), they also use the ```_ = Task.Run()``` but instead of the try catch, they use ```ContinueWith``` .

If you are certain, I'll revert it since it hasn't given a problem so far